server.py

```diff
@@ -14,15 +17,17 @@
     while True:
         client, address = socket.accept()
         client_output = NetworkTerminalOutput(client)
+        client_input = NetworkTerminalInput(client)
```

Comment:
the name could potentially confuse, maybe change these to something like `input_sender`/`output_sender` so people know this isn't really the client but the interface to interact with the client.